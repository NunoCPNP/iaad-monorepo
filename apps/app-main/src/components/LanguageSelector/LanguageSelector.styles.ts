import styled from '@emotion/styled'

export const Container = styled.div`
  ul {
    display: inline-flex;
  }
`
