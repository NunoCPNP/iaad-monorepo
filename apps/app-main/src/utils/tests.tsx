import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

const AllTheProviders = ({ children }) => {
  return <div>{children}</div>
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
