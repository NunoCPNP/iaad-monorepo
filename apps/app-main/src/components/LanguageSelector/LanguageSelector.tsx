import Link from 'next/link'
import Image from 'next/image'
import { useRouter } from 'next/router'

import { Container } from './LanguageSelector.styles'

const LanguageSelector: React.FC = () => {
  const router = useRouter()

  return (
    <Container data-testid="languageselector">
      <ul>
        {router &&
          router.locales.map((locale: string) => (
            <li key={locale}>
              <Link href={router.asPath} locale={locale}>
                <a>
                  <Image src={`/${locale}.svg`} alt={`Switch language to ${locale}`} width={15} height={15} />
                </a>
              </Link>
            </li>
          ))}
      </ul>
    </Container>
  )
}

export default LanguageSelector
