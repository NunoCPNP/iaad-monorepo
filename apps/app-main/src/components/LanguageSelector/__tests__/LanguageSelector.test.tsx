import { render, cleanup } from '../../../utils/tests'

import LanguageSelector from '../LanguageSelector'

afterEach(cleanup)

describe('<LanguageSelector /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<LanguageSelector />)

    const languageselector = queryAllByTestId(/languageselector/i)
    expect(languageselector).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<LanguageSelector />)

    expect(asFragment(<LanguageSelector />)).toMatchSnapshot()
  })
})
