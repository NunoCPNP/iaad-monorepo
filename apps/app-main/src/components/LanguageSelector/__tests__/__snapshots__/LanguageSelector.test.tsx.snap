// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelector /> spec Should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 ul {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 li:not(:last-child) {
  padding-right: 0.5rem;
}

<div>
    <div
      class="emotion-0"
      data-testid="languageselector"
    >
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
